.github/workflows/integration-tests.yml

```diff
@@ -95,14 +95,45 @@ jobs:
         name: ${{ runner.os }}
         path: _release/stack-*
 
+  configuration:
+    name: Check for self-hosted runners
+    runs-on: ubuntu-latest
+    env:
+      CAN_SIGN: ${{ secrets.RELEASE_SIGNING_KEY != '' }}
+    outputs:
+      arm64: ${{ fromJSON(steps.runners.outputs.arm64) }}
+      can-sign: ${{ env.CAN_SIGN }}
+    steps:
+    - name: Check for hosted runners
+      id: runners
+      shell: bash
+      env:
+        SELF_HOSTED_RUNNERS: ${{ secrets.SELF_HOSTED_RUNNERS || (github.repository_owner == 'commercialhaskell' && 'arm64') }}
+      run: |
+        echo "::set-output name=runners::$SELF_HOSTED_RUNNERS"
+        if echo "$SELF_HOSTED_RUNNERS" | grep -q 'arm64'; then
+          echo '::set-output name=arm64::"[\"self-hosted\", \"linux\", \"ARM64\"]"'
```

Comment:
Are the quotation marks causing the problem with the hiatus at:

~~~text
Requested labels: ["self-hosted", "linux", "ARM64"]
Job defined at: commercialhaskell/stack/.github/workflows/integration-tests.yml@refs/pull/5856/merge
Waiting for a runner to pick up this job...
~~~

The original was as follows (no quotation marks):

~~~yaml
linux-arm64:
    name: Linux ARM64
    runs-on: [self-hosted, linux, ARM64]
~~~

Reply:
I'm not sure. I don't have a personal self-hosted runner (I have one at work that I could play w/).

In general, I'd expect `fromJSON` to do more or less the right thing. Since I _think_ it's supposed to be a string or an array per:
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idruns-on

> You can provide `runs-on` as a single string or as an array of strings. If you specify an array of strings, your workflow will run on a self-hosted runner whose labels match all of the specified `runs-on` values, if available. If you would like to run your workflow on multiple machines, use [`jobs.<job_id>.strategy`](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idstrategy).

I _guessed_ that this was the right notation to make it happy.

Let's assume that I guessed wrong.

If it turns out I guessed wrong, I'll file at least one bug against this -- I'll probably start w/ the docs repo, although I feel that this is a bug in the impl...

Reply:
I've made the change - lets see what happens ...